commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -223,6 +224,71 @@ void testReplaceAllComplexScalarFunction() {
         Assertions.assertEquals(objectList, actualList);
     }
 
+    @Test
+    void testReplaceAllComplexConsumer() {
+        List<Complex> data1 = generateList(10);
+        ArrayList<Complex> actual1 = new ArrayList<>();
+        ComplexList expected = new ComplexList();
+        expected.addAll(data1);
+        Assertions.assertThrows(NullPointerException.class, () -> expected.forEach((ComplexConsumer) null));
+        expected.forEach((real, imaginary) -> actual1.add(Complex.ofCartesian(real, imaginary)));
+        Assertions.assertEquals(expected, actual1);
+
+        //Testing case of when list is empty
+        List<Complex> data2 = new ArrayList<>();
+        ArrayList<Complex> actual2 = new ArrayList<>();
+        ComplexList expected2 = new ComplexList();
+        expected2.addAll(data2);
+        expected2.forEach((real, imaginary) -> actual2.add(Complex.ofCartesian(real, imaginary)));
+        Assertions.assertEquals(expected2, actual2);
+    }
+
+    @Test
+    void testGetRealAndImaginary() {
+        ComplexList list = new ComplexList();
+        list.add(Complex.ofCartesian(42, 13));
+        list.addAll(1, list);
+        list.addAll(list);
+        list.set(2, Complex.ofCartesian(200, 1));
+        for (int i = 0; i < list.size(); i++) {
+            Assertions.assertEquals(list.get(i).getReal(), list.getReal(i));
+        }
+        for (int i = 0; i < list.size(); i++) {
+            Assertions.assertEquals(list.get(i).getImaginary(), list.getImaginary(i));
```

Comment:
This can be placed in the same loop as above. Add a simple message to the assertion, e.g. "real".